src/main/java/org/bitcoinj/core/AuxPowHeadersMessage.java

```diff
@@ -0,0 +1,76 @@
+package org.bitcoinj.core;
+
+import org.libdohj.core.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by maxke on 16.10.2015.
+ * Deals with AuxPow headers
+ */
+public class AuxPowHeadersMessage extends HeadersMessage{
+
+    private List<Block> blockHeaders;
+
+    public AuxPowHeadersMessage(NetworkParameters params, byte[] payload) throws ProtocolException {
+        super(params, payload);
+    }
+
+    @Override
+    protected void parse() throws ProtocolException {
+        if (length == UNKNOWN_LENGTH) {
+            int saveCursor = cursor;
+            long numHeaders = readVarInt();
+            cursor = saveCursor;
+
+            // Each header has 80 bytes and one more byte for transactions number which is 00.
+            length = 81 * (int)numHeaders;
+        }
+
+        long numHeaders = readVarInt();
```

Comment:
Does this intentionally read the number of headers twice where an unknown length is provided?